cpp/src/ffi/external_c.cpp

```diff
@@ -0,0 +1,74 @@
+#include "milvus-storage/ffi_c.h"
+
+#include <cstring>
+
+#include <parquet/arrow/reader.h>
+#include <arrow/c/abi.h>
+#include <arrow/c/bridge.h>
+#include <arrow/type_fwd.h>
+
+#include "milvus-storage/ffi_internal/result.h"
+#include "milvus-storage/filesystem/fs.h"
+#include "milvus-storage/properties.h"
+
+FFIResult external_get_file_info(const char* format,
+                                 const char* file_path,
+                                 const Properties* properties,
+                                 uint64_t* out_num_of_rows,
+                                 struct ArrowSchema* out_schema) {
+  milvus_storage::ArrowFileSystemConfig fs_config;
+  milvus_storage::api::Properties properties_map;
+
```

Comment:
check `format`/`file_path`/`properties`/`out_num_of_rows` not NULL before we use it.